src/mono/wasm/runtime/library_mono.js

```diff
@@ -779,6 +779,64 @@ var MonoSupportLib = {
 			// and nested class names like Foo/Bar to Foo.Bar
 			return className.replace(/\//g, '.').replace(/`\d+/g, '');
 		},
+
+		wasm_load_timezone_data: function (file_name) {
+			var binary = read(file_name, "binary");
+			// The timezone file is generated by https://github.com/dotnet/blazor/tree/master/src/TimeZoneData.
+			// The file format of the TZ file look like so
+			//
+			// [4-byte magic number]
+			// [4 - byte length of manifest]
+			// [json manifest]
+			// [data bytes]
+			//
+			// The json manifest is an array that looks like so:
+			//
+			// [...["America/Fort_Nelson",2249],["America/Glace_Bay",2206]..]
+			//
+			// where the first token in each array is the relative path of the file on disk, and the second is the
+			// length of the file. The starting offset of a file can be calculated using the lengths of all files
+			// that appear prior to it.
+			var remainingData = new Uint8Array (binary);
+			var dataview = new DataView(remainingData.buffer);
+			var magic = dataview.getUint32(0, true);
+			//	get magic number
+			if (magic != 0x626c6174) {
+				throw new Error ("File is of wrong type");
+			}
+			var manifestSize = dataview.getUint32(4, true);
+			var manifestContent = Module.UTF8ArrayToString(remainingData, 8, manifestSize);
+			var manifest = JSON.parse(manifestContent.split(/[a-zA-Z//_]+, [0-9]+/g));
+			remainingData = remainingData.slice(manifestSize+8);
+
+			// Create the folder structure
+			// /usr/share/zoneinfo
+			// /usr/share/zoneinfo/Africa
+			// /usr/share/zoneinfo/Asia
+			// ..
+			FS.mkdir("usr");
+			Module['FS_createPath']('/', 'usr', true, true);
+			FS.mkdir("share");
+			Module['FS_createPath']('/usr', 'share', true, true);
+			FS.mkdir("/usr/share/zoneinfo");
```

Comment:
if the prefix is passed to the method you can just prepend it to m[0] and you can remove this extra code and it will be created in the folder generation

Reply:
Hm, it doesn't seem to create /usr/share/zoneinfo during folder generation:
```
folders.forEach(folder => {
		FS.mkdir(`${prefix}/${folder}`)
		Module['FS_createPath'](prefix, folder, true, true);
		});
```